src/Filterable.ts

```diff
@@ -108,3 +109,129 @@ export interface Filterable3C<F extends URIS3, U, L> extends Functor3C<F, U, L>,
   readonly filterMap: <A, B>(fa: Type3<F, U, L, A>, f: (a: A) => Option<B>) => Type3<F, U, L, B>
   readonly filter: <A>(fa: Type3<F, U, L, A>, p: Predicate<A>) => Type3<F, U, L, A>
 }
+
+export function span<F extends URIS3, U, L>(
```

Comment:
I'm not sure you can define `span`, `takeWhile` and `dropWhile` for a generic `Filterable` since there's no order involved: what's a sensible notion of "longest initial substructure"?

Reply:
Well, makes sense. Somehow I didn't think about that :)